drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -477,30 +554,21 @@ static const struct stm32_pclken rtc_clk[] = STM32_DT_INST_CLOCKS(0);
 static const struct rtc_stm32_config rtc_config = {
 	.counter_info = {
 		.max_top_value = UINT32_MAX,
-		.freq = 1,
+#ifndef CONFIG_COUNTER_RTC_STM32_SUBSECONDS
+		/* freq = 1Hz for not subsec based driver */
+		.freq = RTCCLK_FREQ / ((RTC_ASYNCPRE + 1) * (RTC_SYNCPRE + 1)),
+#else /* !CONFIG_COUNTER_RTC_STM32_SUBSECONDS */
+		.freq = RTCCLK_FREQ / (RTC_ASYNCPRE + 1),
+#endif /* CONFIG_COUNTER_RTC_STM32_SUBSECONDS */
```

Comment:
I might have missed something but in BCD mode, freq is always `RTCCLK_FREQ / ((RTC_ASYNCPRE + 1) * (RTC_SYNCPRE + 1))`

Reply:
It depends which freq you mean :-) the subsecond register is clocked with signal after only one prescaler - the async one. So when we are using subsecond as a base, this freq is used as a reference.